src/test/fuzz/parse_iso8601.cpp

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) 2019 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <util/time.h>
+
+#include <cassert>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+void test_one_input(const std::vector<uint8_t>& buffer)
+{
+    FuzzedDataProvider fuzzed_data_provider(buffer.data(), buffer.size());
+
+    const int64_t random_time = fuzzed_data_provider.ConsumeIntegral<int64_t>();
+    const std::string random_string = fuzzed_data_provider.ConsumeRemainingBytesAsString();
+
+    const std::string iso8601_datetime = FormatISO8601DateTime(random_time);
```

Comment:
This will fail when the year overflows after adding `1900`. E.g.

```
$ base64 crash-59af3ac3c0a4395f770dbbd0f956cd680aa65a3d 
AAAAALQ3CVA=

Reply:
@MarcoFalke I've seen that case. I'm running with this patch in my local tree:

```diff
diff --git a/src/util/time.cpp b/src/util/time.cpp
index 2afff2626..c311c6a49 100644
--- a/src/util/time.cpp
+++ b/src/util/time.cpp
@@ -98,6 +98,14 @@ std::string FormatISO8601DateTime(int64_t nTime) {
 #else
     gmtime_r(&time_val, &ts);
 #endif
+    if (!((ts.tm_year >= 0 && ts.tm_year <= 9999-1900) &&
+          (ts.tm_mon >= 0 && ts.tm_mon <= 11) &&
+          (ts.tm_mday >= 1 && ts.tm_mday <= 31) &&
+          (ts.tm_hour >= 0 && ts.tm_hour <= 24) &&
+          (ts.tm_min >= 0 && ts.tm_min <= 59) &&
+          (ts.tm_sec >= 0 && ts.tm_sec <= 59))) {
+        return {};
+    }
     return strprintf("%04i-%02i-%02iT%02i:%02i:%02iZ", ts.tm_year + 1900, ts.tm_mon + 1, ts.tm_mday, ts.tm_hour, ts.tm_min, ts.tm_sec);
 }
 
@@ -109,6 +117,11 @@ std::string FormatISO8601Date(int64_t nTime) {
 #else
     gmtime_r(&time_val, &ts);
 #endif
+    if (!((ts.tm_year >= 0 && ts.tm_year <= 9999-1900) &&
+          (ts.tm_mon >= 0 && ts.tm_mon <= 11) &&
+          (ts.tm_mday >= 1 && ts.tm_mday <= 31))) {
+          return {};
+    }
     return strprintf("%04i-%02i-%02i", ts.tm_year + 1900, ts.tm_mon + 1, ts.tm_mday);
 }
```

Perhaps worth submitting upstreams?